<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SegmentationUNet</class>
 <widget class="qMRMLWidget" name="SegmentationUNet">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>434</width>
    <height>672</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="basicCollapsibleButton">
     <property name="text">
      <string>Input/output</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Trained model:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkPathLineEdit" name="modelPathLineEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="filters">
         <set>ctkPathLineEdit::Files|ctkPathLineEdit::NoDot|ctkPathLineEdit::NoDotDot|ctkPathLineEdit::Readable</set>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>ctkPathLineEdit::AdjustToMinimumContentsLength</enum>
        </property>
        <property name="minimumContentsLength">
         <number>32</number>
        </property>
       </widget>
      </item>
      <!-- <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Input sequence browser:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLNodeComboBox" name="browserSelector">
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLSequenceBrowserNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="hideChildNodeTypes">
         <stringlist notr="true"/>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="interactionNodeSingletonTag">
         <string notr="true"/>
        </property>
       </widget>
      </item> -->
      <item row="3" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Input volume:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="qMRMLNodeComboBox" name="inputSelector">
        <property name="toolTip">
         <string>Pick the input to the algorithm.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Prediction volume:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="qMRMLNodeComboBox" name="outputSelector">
        <property name="toolTip">
         <string>Pick the output to the algorithm.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QCheckBox" name="checkBox_sam">
          <property name="text">
           <string>SAM</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_unext">
          <property name="text">
           <string>UNext</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Model:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="qMRMLSegmentEditorWidget" name="SegmentEditorWidget">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="defaultTerminologyEntrySettingsKey">
      <string notr="true"/>
     </property>
     <property name="jumpToSelectedSegmentEnabled">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <!-- <item>
    <widget class="QPushButton" name="embeddingButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>Compute embedding</string>
     </property>
    </widget>
   </item> -->
   <item>
    <widget class="QPushButton" name="applyButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Run the algorithm.</string>
     </property>
     <property name="text">
      <string>Apply segmentation</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="reconButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>Apply reconstruction</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="startLiveReconButton">
     <property name="text">
      <string>Start live reconstruction</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="stopLiveReconButton">
     <property name="text">
      <string>Pause live reconstruction</string>
     </property>
    </widget>
   </item>   <item>
    <widget class="QLabel" name="feedbackLabel">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkPathLineEdit</class>
   <extends>QWidget</extends>
   <header>ctkPathLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSegmentEditorWidget</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLSegmentEditorWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>SegmentationUNet</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SegmentationUNet</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SegmentationUNet</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>browserSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SegmentationUNet</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>SegmentEditorWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>216</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
